### DISCUSSION TITLE
How to do customized templates from preinstalled ones?

### DISCUSSION THREAD

**Discussion by quanta-science** [getnikola/nikola#3581] [Q&A] [answered]
Hi guys, 
I tried to apply another template (not default base.tmpl) for a dozen of my pages.
I did the following.
i. created copy of base.tmpl, renamed it to base1.tmpl and did some minor changes in it;
ii. created test page in HTML input format and added "..template: base1.tmpl" in its meta section;
iii. after "nikola build" I found that there is no any content in output test page though I put some text in input test page.

I use JIDN (mako) theme.

I very appreciate your help.

**Comment by Kwpolska** [comment]
Nikola templates have a hierarchy to them. The default template for posts is `post.tmpl`, and it inherits from `base.tmpl`. This means that if you want a custom `base1.tmpl` for certain posts, you also need `post1.tmpl` — which would inherit from the new base, and the post would also need to specify `.. template: post1.tmpl`.

**Comment by ralsina** [comment]
On Tue, Oct 12, 2021 at 8:29 AM Chris Warrick ***@***.***>
wrote:

> Nikola templates have a hierarchy to them. The default template for posts
> is post.tmpl, and it inherits from base.tmpl. This means that if you want
> a custom base1.tmpl for certain posts, you also need post1.tmpl — which
> would inherit from the new base, and the post would also need to specify ..
> template: post1.tmpl.
>
>
In most cases just the post1.tmpl will be enough. If there's anything you
need changed on base.tmpl you can override it in post.tmpl too

**Reply by quanta-science** [reply]
What to do in case of page.tmpl? as I've look through conf.py, pages are generated on the base of page.tmpl

**Comment by ralsina** [comment, marked as answer]
On Tue, Oct 12, 2021 at 11:07 AM quanta-science ***@***.***>
wrote:

> What to do in case of page.tmpl? as I've look through conf.py, pages are
> generated on the base of page.tmpl
>

Same thing, start off a copy of page.tmpl and if there's any block in
another template you want to override it can be done there too.

**Reply by quanta-science** [reply]
Thank you guys very much! It worked. I had to copy all the files from which the template was inherited.

**Comment by quanta-science** [comment]
Hi guys,
Hope everyone of you is ok. Since 1 year I have started my static site everything was ok. And now I would like to add some dynamic content to my site. That is why I have a question to you. Is it possible to add some scripts and data file to /files-folder to make user registration (sign-up, sign-in, cabinets, etc) on Nikola static site?

Kind regards
Murat

**Comment by quanta-science** [comment]
I forgot my 2nd question. Is there any method to make bilingual website on Nikola static site?

**Reply by Kwpolska** [reply]
Nikola has support for multiple languages in one site and translating posts. You can configure it in conf.py (`TRANSLATIONS`).

**Comment by ralsina** [comment]
Please don't start unrelated questions in old issues :-)

Other than that... Maybe? You could setup something like PHP or whatever,
but I have no idea of the specifics.
